os/fs/smartfs/smartfs_utils.c

```diff
@@ -521,6 +521,253 @@ int set_used_byte_count(uint8_t *used, uint16_t count)
 	return OK;
 }
 #endif
+
+/****************************************************************************
+ * Name: smartfs_seek_internal
+ *
+ * Description: Performs the logic of the seek function.  This is an internal
+ *              function because it does not provide semaphore protection and
+ *              therefore must be called from one of the other public
+ *              interface routines (open, seek, etc.).
+ *
+ ****************************************************************************/
+
+off_t smartfs_seek_internal(struct smartfs_mountpt_s *fs, struct smartfs_ofile_s *sf, off_t offset, int whence)
+{
+	struct smart_read_write_s readwrite;
+	struct smartfs_chain_header_s *header;
+	int ret;
+	off_t newpos;
+	off_t sectorstartpos;
+#ifdef CONFIG_SMARTFS_DYNAMIC_HEADER
+	int sector_used = 0;
+#endif
+	/* Test if this is a seek to get the current file pos */
+
+	if ((whence == SEEK_CUR) && (offset == 0)) {
+		return sf->filepos;
+	}
+
+	/* Test if we need to sync the file */
+
+	if (sf->byteswritten > 0) {
+		/* Perform a sync */
+
```

Comment:
same like above can you also remove this?

Reply:
done